In [1]:
# Step 1: Import dependencies

import matplotlib.pyplot as plt
from sklearn.neighbors import KNeighborsClassifier
import pandas as pd
import numpy as np

In [2]:
# Step 2: grab the data

df = pd.read_csv('resources/diabetes.csv')
df.head()

Pregnancies  Glucose  BloodPressure  SkinThickness  Insulin   BMI  \
0            6      148             72             35        0  33.6   
1            1       85             66             29        0  26.6   
2            8      183             64              0        0  23.3   
3            1       89             66             23       94  28.1   
4            0      137             40             35      168  43.1   

   DiabetesPedigreeFunction  Age  Outcome  
0                     0.627   50        1  
1                     0.351   31        0  
2                     0.672   32        1  
3                     0.167   21        0  
4                     2.288   33        1

In [4]:
# Step 3: Assign your X and y (features/data and outcome/target variable)

# start with the y (outcome/target variable)
target = df["Outcome"]

# choose target names as well because the outcome is in binary form
target_names = ['negative', 'positive']

In [5]:
# now choose the X or features and data

data = df.drop("Outcome", axis=1)
feature_names = data.columns
data.head()

Pregnancies  Glucose  BloodPressure  SkinThickness  Insulin   BMI  \
0            6      148             72             35        0  33.6   
1            1       85             66             29        0  26.6   
2            8      183             64              0        0  23.3   
3            1       89             66             23       94  28.1   
4            0      137             40             35      168  43.1   

   DiabetesPedigreeFunction  Age  
0                     0.627   50  
1                     0.351   31  
2                     0.672   32  
3                     0.167   21  
4                     2.288   33

In [6]:
# Step 4: Employ train test split and make the training and testing sets

from sklearn.model_selection import train_test_split

X_train, X_test, y_train, y_test = train_test_split(data, target, random_state=42)

In [7]:
# Step 5: create the SVM model

from sklearn.svm import SVC
model = SVC(kernel='linear')

In [8]:
# Step 6: train the model with the training dataset

model.fit(X_train, y_train)

SVC(kernel='linear')

In [9]:
# Step 7: make predictions on the model using the testing data

predictions = model.predict(X_test)

In [10]:
# Step 8: Build the classification report

from sklearn.metrics import classification_report
print(classification_report(y_test, predictions, target_names=target_names))

              precision    recall  f1-score   support

    negative       0.79      0.78      0.79       123
    positive       0.62      0.64      0.63        69

    accuracy                           0.73       192
   macro avg       0.71      0.71      0.71       192
weighted avg       0.73      0.73      0.73       192



<strong>Precision</strong>:  
* Out of 123 claims that the test data is negative of diabetes, 79% was correct
* Out of 69 claims that the test data is positive of diabetes, 62% was correct  

<strong>Recall</strong>:  
* Out of all the negative outcomes in the test set, 78% was captured by the model
* Out of all the positive outcomes in the test set, 64% was captured by the model  

<strong>f1_score</strong>:  
* The average 'strength' of the model to predict negative results is 79%
* the average 'strength' of the model to predict positive results is 63%  

<strong>Overall:</strong>
* If we rely on accuracy, the model is 73% correct